cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillController.java

```diff
@@ -484,7 +475,118 @@ private void deletePVCs() {
     return resultFuture;
   }
 
+  /**
+   * Deletes the job controlled by this controller asynchronously.
+   *
+   */
+  private void deleteJob() {
+    apiClient.setDebugging(true);
+    BatchV1Api batchApi = new BatchV1Api(apiClient);
+
+    String name = meta.getName();
+    try {
+      batchApi.deleteNamespacedJobAsync(name, kubeNamespace, null, null, null, null, null,
+                                        new V1DeleteOptions(), new ApiCallbackAdapter<V1Status>() {
+          @Override
+          public void onFailure(ApiException e, int statusCode, Map<String, List<String>> responseHeaders) {
```

Comment:
do we also clean up pods if the job fails?

Reply:
Hmm with new k8s client the dependent pods does get deleted when job is deleted.

Reply:
Refactored logic to rely on k8s to delete the dependent pods.